src/lib/Server/Input/Parser/AbstractDestinationLocationParser.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace Ibexa\Rest\Server\Input\Parser;
+
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\Values\Content\Location;
+use Ibexa\Contracts\Rest\Exceptions;
+use Ibexa\Contracts\Rest\Input\ParsingDispatcher;
+use Ibexa\Rest\Input\BaseParser;
+use Ibexa\Rest\Server\Validation\Builder\Input\Parser\BaseInputParserValidatorBuilder;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+abstract class AbstractDestinationLocationParser extends BaseParser
+{
+    public const string DESTINATION_KEY = 'destination';
+
+    public function __construct(
+        private readonly LocationService $locationService,
+        protected readonly ValidatorInterface $validator,
+    ) {
+    }
+
+    /**
+     * @phpstan-param array{
+     *     'destination': string,
+     * } $data
+     *
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     * @throws \Ibexa\Contracts\Rest\Exceptions\Parser
+     */
+    public function parse(array $data, ParsingDispatcher $parsingDispatcher): Location
+    {
+        $this->validateInputData($data);
+
+        return $this->getLocationByPath($data[self::DESTINATION_KEY]);
+    }
+
+    /**
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    private function getLocationByPath(string $path): Location
+    {
+        return $this->locationService->loadLocation(
+            $this->extractLocationIdFromPath($path)
+        );
+    }
+
+    private function extractLocationIdFromPath(string $path): int
+    {
+        $pathParts = explode('/', $path);
+
+        return (int)array_pop($pathParts);
```

Comment:
This assumes that the last part of the path string is integerish.

While trying to load `Location` with ID 0 will result in an error, it will not be clear to the end-user why a Location with ID 0 was tried to be loaded.

You can use https://github.com/webmozarts/assert?tab=readme-ov-file#assertions `integerish` check to ensure that we convert a valid value.

Reply:
We don't use this package in `ibexa/rest`.

Reply:
I don't see a problem with adding it, Paweł's remark seems valid to me.